helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/algorithm/RebalanceAlgorithm.java

```diff
@@ -20,25 +20,27 @@
  */
 
 import org.apache.helix.controller.rebalancer.waged.model.ClusterModel;
-import org.apache.helix.model.ResourceAssignment;
-
-import java.util.Map;
+import org.apache.helix.controller.rebalancer.waged.model.OptimalAssignment;
 
 /**
- * A generic rebalance algorithm interface for the WAGED rebalancer.
+ * A generic interface used to give the optimal assignment given the runtime cluster environment.
  *
- * @see <a href="Rebalance Algorithm">https://github.com/apache/helix/wiki/Design-Proposal---Weight-Aware-Globally-Even-Distribute-Rebalancer#rebalance-algorithm-adapter</a>
+ * <pre>
+ * @see <a href="https://github.com/apache/helix/wiki/
+ * Design-Proposal---Weight-Aware-Globally-Even-Distribute-Rebalancer
+ * #rebalance-algorithm-adapter">Rebalance Algorithm</a>
+ * </pre>
  */
 public interface RebalanceAlgorithm {
+  enum Type {
+    CONSTRAINTS
```

Comment:
I don't think we need this type. In my opinion, each algorithm class shall have their own factory. Or if you really have a valid case that we need to support multiple algorithms in the default factory, let's put the type in the factory instead of the interface.